server/task/task-memory/src/test/java/org/apache/james/task/SerialTaskManagerWorkerTest.java

```diff
@@ -225,6 +225,10 @@ void theWorkerShouldCancelAnInProgressTask() throws InterruptedException {
 
         resultMono.block(Duration.ofSeconds(10));
 
+        // Due to the use of signals, cancellation cannot be instantaneous
+        // Let a grace period for the cancellation to complete to increase test stability
+        Thread.sleep(50);
```

Comment:
i am not convinced by this fix as we block on the result Mono the line before.
And in the code of the SerialTaskManager when the result of the task is returned the `listener.cancelled` method is already called.

so i see not point in adding a wait since after the block the listener.cancelled would have been called.
Furthermore when launching this test until failure with intellij it doesn't fail on my local machine even after 3000+ runs.

Maybe the instabiblity is coming from another test ?
How did you reproduced the issue ?

Reply:
> i am not convinced by this fix as we block on the result Mono the line before.

Well it don't change task cancelation is propagated by a signal. You can't expect the signal to be reacted upon when your block finishes.

> so i see not point in adding a wait since after the block the listener.cancelled would have been called.

This is more to leverage a limitation of mockito `atLeastOnce` call

```
Concider the following scenario:

 - t0 : cancelled is triggered
 - t1 = t0 + delta1 : block finishes
 - t2 = t1 + delta2 : atLeast once check is performed
 - t3 = t2 + delta3 : cancelled is triggered (again)
 - t4 = t3 + delta3: we check no more interaction is performed with the mock

t4 check fails however the business rule (no more interactions with the mock, but as many cancels as you want) is not violated.
```

IMO we need to work at better enforcing the business rule.

My proposal is a way to do it.

Sorry but I don't want to invest in some mockito dark magic.

> Furthermore when launching this test until failure with intellij it doesn't fail on my local machine even after 3000+ runs.
> How did you reproduced the issue ?

Funny I reproduced it fairly reliably, in isolation with a `@RepeatedTest`. Easily.

Reply:
I can't make it fails on my local machine even with `@RepeatedTest`.
But if this fix the issue you can reproduce, it's good for me.